apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDataReader.cs

```diff
@@ -254,6 +254,24 @@ public async Task<int> GetOrdinalAsync(string name, CancellationToken cancellati
             return _fieldIndex[name];
         }
 
+        /// <summary>
+        /// Gets the read timstamp if <see cref="TimestampBound.ReturnReadTimestamp" /> was specified.
+        /// </summary>
+        /// <remarks>
+        /// The read timestamp is only available after at least one result is
+        /// read. If the returned results are empty, then a read timestamp
+        /// won't be available.
+        /// </remarks>
+        /// <returns>The value converted to a <see cref="Timestamp"/>.</returns>
+        /// <exception cref="InvalidOperationException">A read timestamp is unavailable,
+        /// possibly becuase <see cref="TimestampBound.ReturnReadTimestamp" /> was not set.</exception>
+        public Timestamp GetReadTimestamp()
+        {
+            GaxPreconditions.CheckState(PopulateMetadata().Transaction != null, "Transaction unavailable so read timestamp can't be returned.");
```

Comment:
Can we call PopulateMetadata once and keep the result in a local variable? (I know it's cached, but it feels cleaner to me.)